GVFS/GVFS.Installer.Windows/Setup.iss

```diff
@@ -616,6 +616,70 @@ begin
   Result := True;
 end;
 
+type
+  UpgradeRing = (urUnconfigured, urNone, urFast, urSlow);
+
+function GetConfiguredUpgradeRing(): UpgradeRing;
+var
+  ResultCode: integer;
+  ResultString: ansiString;
+begin
+  Result := urUnconfigured;
+  if ExecWithResult('gvfs.exe', 'config upgrade.ring', '', SW_HIDE, ewWaitUntilTerminated, ResultCode, ResultString) then begin
+    if ResultCode = 0 then begin
+      ResultString := AnsiLowercase(Trim(ResultString));
+      Log('GetConfiguredUpgradeRing: upgrade.ring is ' + ResultString);
+      if CompareText(ResultString, 'none') = 0 then begin
+        Result := urNone;
+      end else if CompareText(ResultString, 'fast') = 0 then begin
+        Result := urFast;
+      end else if CompareText(ResultString, 'slow') = 0 then begin
+        Result := urSlow;
+      end else begin
+        Log('GetConfiguredUpgradeRing: Unknown upgrade ring: ' + ResultString);
+      end;
+    end else begin
+      Log('GetConfiguredUpgradeRing: Call to gvfs config upgrade.ring failed with ' + SysErrorMessage(ResultCode));
+    end;
+  end else begin
+    Log('GetConfiguredUpgradeRing: Call to gvfs config upgrade.ring failed with ' + SysErrorMessage(ResultCode));
+  end;
+end;
+
+procedure SetNuGetFeedIfNecessary();
+var
+  ConfiguredRing: UpgradeRing;
+  RingName: String;
+  TargetFeed: String;
+  FeedPackageName: String;
+  ResultCode: Integer;
+  ResultString: ansiString;
+begin
+  ConfiguredRing := GetConfiguredUpgradeRing();
+  if ConfiguredRing = urFast then begin
+    RingName := 'Fast';
+  end else if (ConfiguredRing = urSlow) or (ConfiguredRing = urNone) then begin
+    RingName := 'Slow';
+  end else begin
+    Log('SetNuGetFeedIfNecessary: No upgrade ring configured. Not configuring NuGet feed.')
+    exit;
+  end;
+
+  TargetFeed := Format('https://pkgs.dev.azure.com/microsoft/_packaging/VFSForGit-%s/nuget/v3/index.json', [RingName]);

```

Comment:
Don't need to set if this is already configured.

Reply:
(both feed and packagename)

Reply:
I suppose we could return early if the nuget feed (and/or package) is already set.

Reply:
One possible advantage of setting it every time is that if someone changes their ring without changing their feed this will fix it for them (not sure if that's something we want to do though).